core/src/main/java/org/infinispan/configuration/cache/AbstractStoreConfigurationBuilder.java

```diff
@@ -189,7 +189,7 @@ private void validateStoreAttributes() {
             throw CONFIG.sharedStoreWithLocalCache();
 
          if (!preload && builder.indexing().enabled()
-               && builder.indexing().indexLocalOnly() && !getBuilder().template())
+               && builder.indexing().index() == Index.PRIMARY_OWNER && !getBuilder().template())
```

Comment:
call `isPrimaryOwner` method

Reply:
But I've just deprecated Index.isPrimaryOwner. All those isX methods in an enum that are equivalent to a simple == X are useless. Why not add that clutter for all enum members ? Why do we even have enums? :)   But there are exceptions where such methods make sense, like when they test that the enum member is part of a subset of the whole set. Index.isEnabled is such a legitimate case. Everything else is clutter and abuse.

Reply:
I didn't see that deprecation, cool then